source/collimation.f90

```diff
@@ -3192,7 +3192,7 @@ subroutine collimate_end_collimator()
   real(kind=fPrec) hdfx,hdfxp,hdfy,hdfyp,hdfdee,hdfs
 #endif
 
-  real(kind=fPrec) stracki
+  ! real(kind=fPrec) stracki ! stracki makes no sense here
```

Comment:
not sure about this... I think it is important to keep it, and maybe it should have been in the subroutine interface.
Is there any chance that the compilers were actually taking the value from the parent subroutine?

Reply:
as discussed, this subroutine does not even need `stracki`, and anyway a non-zero long collimator as from `fort.2` is caught earlier in the code and the code will immediately exit

Reply:
Also, the `stracki` variable is never set in this routine, so the behaviour is unpredictable.